src/helpers/helpers.dom.ts

```diff
@@ -222,9 +222,6 @@ export function retinaScale(
   const deviceHeight = Math.floor(chart.height * pixelRatio);
```

Comment:
The deviceHeight could end up being a little bit smaller now. For example if devicePixelRatio is 1.75, and the `newSize` in core.controller has a width of 15.9, the deviceWidth will end up being 26. (`Math.floor((Math.floor(15.9)*1.75)`), when with current implementation it ends up being 27 (`Math.floor(15.9*1.75)`).

For this reason, I would revert the change in core.controller and replace the `chart.height = deviceHeight / pixelRatio` with `chart.height = Math.floor(chart.height)` below. (width obviously too)